### PR TITLE
ARC-8: Fix GitHub Actions for Jira mapping.

### PR DESCRIPTION
**Pull request by atrigueiro**

### CODE REVIEW

File: lib/transforms/deployment.js [hunk 1]
```diff
@@ -44,7 +40,6 @@ function mapEnvironment({ environment }, isProdEnvironment) {
   const jiraEnv = Object.keys(environmentMapping).find(key => isEnvironment(environmentMapping[key]));
 
   if (!jiraEnv) {
-    app.log.info(`Unmapped environment detected for deployment. Unmapped value is ${environment}. Sending it as unmapped to Jira.`);
```
Comment (atrigueiro) [inline lib/transforms/deployment.js:47]: Couldn't access `app` inside this function, so moved logging to use `context` below.

File: lib/transforms/workflow.js [hunk 2]
```diff
@@ -52,19 +66,14 @@ module.exports = (payload) => {
         state: mapStatus(workflow_run),
         lastUpdated: workflow_run.updated_at,
         issueKeys,
-        references: workflow_run.pull_requests.map(pr => (
```
Comment (atrigueiro) [inline lib/transforms/workflow.js:55]: `map` throws an error is `pull_requests` is empty. Also, `references` can't be an empty array, so moved this check to the bottom.

File: lib/transforms/workflow.js [hunk 3]
```diff
@@ -30,15 +30,29 @@ function mapStatus({ status, conclusion }) {
...
+        name: pr.head.ref,
+        uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
+      },
+    }
+  ));
+}
+module.exports = async (payload) => {
   const { workflow_run } = payload;
-  const { issueKeys } = parseSmartCommit(workflow_run.head_branch);
+  const { issueKeys } = parseSmartCommit(`${workflow_run.head_branch}\n${workflow_run.head_commit.message}`);
```
Comment (atrigueiro) [inline lib/transforms/workflow.js:49]: Get issue key from commit message, as well as the branch name.

File: lib/transforms/deployment.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ module.exports = async (context) => {
         environment: {
           id: deployment_status.environment,
           displayName: deployment_status.environment,
-          type: mapEnvironment(deployment_status, deployment.production_environment),
```
Comment (atrigueiro) [inline lib/transforms/deployment.js:83]: `production_environment` is not a valid property of the `deployment` object.

**Review by mboudreau — APPROVED** [review]

**Pull request merged**